src/services/shopify-client.ts

```diff
@@ -0,0 +1,25 @@
+export enum SubscriberListState {
+  ready,
+  synchronizing_contacts,
+}
+
+export interface SubscriberList {
+  name: string;
+  id: number;
+  amount_subscribers: number;
+  state: SubscriberListState;
+}
+
+export interface ShopifyConnection {
+  shop_name: string;
+  synchronization_date: Date | null;
+  list: SubscriberList;
+  connected: boolean;
+}
+
+export interface ShopifyClient {
+  getShopifyData(email: string, apiKey: string): Promise<ShopifyConnection>;
+  disconnect(email: string, apikey: string): Promise<boolean>;
```

Comment:
Why does it return a boolean? If the idea is return _false_ when there is an error, I would suggest ussing something like [EmptyResult<TError>](https://github.com/FromDoppler/doppler-webapp/blob/master/src/services/doppler-legacy-client.ts#L19).

Maybe those classes could be moved to a common place.

Reply:
About the EmptyResult, yes maybe it's better. Which classes? I don't get you with this.